l10n_it_asset_management/models/asset_depreciation.py

```diff
@@ -0,0 +1,606 @@
+    )
+
+    amount_gain = fields.Monetary(
+        compute="_compute_amounts",
+        string="Capital Gain",
+        store=True,
+    )
+
+    amount_historical = fields.Monetary(
+        compute="_compute_amounts",
+        store=True,
+        string="Historical Amount",
+    )
+
+    amount_in = fields.Monetary(
+        compute="_compute_amounts",
+        store=True,
+        string="In Amount",
+    )
+
+    amount_loss = fields.Monetary(
+        compute="_compute_amounts",
+        store=True,
+        string="Capital Loss",
+    )
+
+    amount_out = fields.Monetary(
+        compute="_compute_amounts",
+        store=True,
+        string="Out Amount",
+    )
+
+    amount_residual = fields.Monetary(
+        compute="_compute_amounts",
+        store=True,
+        string="Residual Amount",
+    )
+
+    asset_id = fields.Many2one(
+        "asset.asset",
+        ondelete="cascade",
+        readonly=True,
+        required=True,
+        string="Asset",
+    )
+
+    base_coeff = fields.Float(
+        default=1,
+        help="Coeff to compute amount depreciable from purchase amount",
+        string="Depreciable Base Coeff",
+    )
+
+    company_id = fields.Many2one(
+        "res.company", readonly=True, related="asset_id.company_id", string="Company"
+    )
+
+    currency_id = fields.Many2one(
+        "res.currency", readonly=True, related="asset_id.currency_id", string="Currency"
+    )
+
+    date_start = fields.Date()
+
+    dismiss_move_id = fields.Many2one("account.move", string="Dismiss Move")
+
+    first_dep_nr = fields.Integer(
+        default=1,
+        string="First Dep. Num",
+    )
+
+    force_all_dep_nr = fields.Boolean(string="Force All Dep. Num")
+
+    force_first_dep_nr = fields.Boolean(string="Force First Dep. Num")
+
+    last_depreciation_date = fields.Date(
+        compute="_compute_last_depreciation_date",
+        store=True,
+        string="Last Dep.",
+    )
+
+    line_ids = fields.One2many(
+        "asset.depreciation.line", "depreciation_id", string="Lines"
+    )
+
+    mode_id = fields.Many2one(
+        "asset.depreciation.mode",
+        required=True,
+        string="Mode",
+    )
+
+    percentage = fields.Float(string="Depreciation (%)")
+
+    pro_rata_temporis = fields.Boolean(string="Pro-rata Temporis")
+
+    requires_account_move = fields.Boolean(
+        readonly=True, related="type_id.requires_account_move"
+    )
+
+    state = fields.Selection(
+        [
+            ("non_depreciated", "Non Depreciated"),
+            ("partially_depreciated", "Partially Depreciated"),
+            ("totally_depreciated", "Depreciated"),
+        ],
+        compute="_compute_state",
+        default="non_depreciated",
+        store=True,
+    )
+
+    type_id = fields.Many2one("asset.depreciation.type", string="Depreciation Type")
+
+    zero_depreciation_until = fields.Date(string="Zero Depreciation Up To")
+
+    @api.model
+    def create(self, vals):
```

Comment:
Dovresti usare il decoratore `@api.model_create_multi` e modificare la funzione per warning is` not overriding the create method in batch`

Reply:
Fatto